set.mm

```diff
@@ -542114,6 +542158,237 @@ sethood hypotheses (compare ~ opth ).  (Contributed by BJ, 6-Oct-2018.) $)
+
+  ${
+    $d x A $.
+    $( Utility theorem for the inverse of ` inftyexpi ` .  (Contributed by BJ,
+       22-Jun-2019.) $)
+    bj-inftyexpiinv $p |- ( A e. ( -u _pi (,] _pi ) ->
+                                           ( 1st ` ( inftyexpi ` A ) ) = A ) $=
+      ( vx cpi cneg cioc co wcel cinftyexpi cfv cc cop cv opeq1 df-bj-inftyexpi
+      c1st opex fvmpt fveq2d cvv wceq cnex op1stg mpan2 eqtrd ) ACDCEFZGZAHIZOI
+      AJKZOIZAUFUGUHOBABLZJKUHUEHUJAJMBNAJPQRUFJSGUIATUAAJUESUBUCUD $.
+  $}
+
+  ${
+    $d A x $.
+    $( An element of the circle at infinity is not a complex number.
+       (Contributed by BJ, 22-Jun-2019.) $)
+    bj-inftyexpidisj $p |- -. ( inftyexpi ` A ) e. CC $=
+      ( vx cinftyexpi cdm wcel cfv cc wn cop wceq cpi opex cvv cnex 0ncn mtbiri
+      c0 syl pm2.61i eleq1d cneg cioc co cv opeq1 df-bj-inftyexpi dmmpti eleq2s
+      fvmpt cpr prid2 csn wo eqid olci wb bj-elopg mpan2 mpbiri bnj1224 sylancr
+      en3lp opprc1 eleq1 eqcom biimpi mtbii ndmfv ) ACDZEZACFZGEZHZVJVKAGIZJZVM
+      VOAKUAKUBUCZVIBABUDZGIZVNVPCVQAGUEBUFZAGLUIBVPVRCVQGLVSUGUHVOVNGEZVLAMEZV
+      THZWAGAGUJZEZWCVNEZWBAGNUKWAWEWCAULJZWCWCJZUMZWGWFWCUNUOWAGMEWEWHUPNAGWCM
+      MUQURUSWDWEVTGWCVNVBUTVAWAHVNQJZWBAGVCWIVTQGEZOVNQGVDPRSVOVNVKGVOVNVKJVKV
+      NVEVFTVGRVJHZVLWJOWKVKQGACVHTPS $.
+  $}
+
+  $( Syntax for the circle at infinity ` CCinfty ` . $)
+  cccinfty $a class CCinfty $.
+
+  $( Definition of the circle at infinity ` CCinfty ` .  (Contributed by BJ,
+     22-Jun-2019.) $)
+  df-bj-ccinfty $a |- CCinfty = ran inftyexpi $.
+
+  ${
+    $d x y $.
+    $( The circle at infinity is disjoint form the set of complex numbers.
+       (Contributed by BJ, 22-Jun-2019.) $)
+    bj-ccinftydisj $p |- ( CC i^i CCinfty ) = (/) $=
+      ( vx vy vz cccinfty cin wcel cinftyexpi cfv wex bj-inftyexpidisj nex wceq
+      cc cv wa elin crn cdm cpi syl wrex wfun cneg cioc df-bj-inftyexpi funmpt2
+      wi co cop elrnrexdm ax-mp rexex df-bj-ccinfty eleq2s anim2i sylbi exancom
+      ancom 19.41v bitri sylbb2 eleq1 biimpac eximi mto bj-nel0 ) AMDEZANZVGFZB
+      NZGHZMFZBIZVLBVJJKVIVHMFZVHVKLZOZBIZVMVIVNVOBIZOZVQVIVNVHDFZOVSVHMDPVTVRV
+      NVRVHGQZDVHWAFZVOBGRZUAZVRGUBWBWDUGCSUCSUDUHCNMUIGCUEUFBGVHUJUKVOBWCULTUM
+      UNUOUPVSVRVNOZVQVNVRURVQVOVNOBIWEVNVOBUQVOVNBUSUTVATVPVLBVOVNVLVHVKMVBVCV
+      DTVEVF $.
+  $}
+
+  $( Syntax for the set of extended complex numbers ` CCbar ` . $)
+  cccbar $a class CCbar $.
+
+  $( Definition of the set of extended complex numbers ` CCbar ` .
+     (Contributed by BJ, 22-Jun-2019.) $)
+  df-bj-ccbar $a |- CCbar = ( CC u. CCinfty ) $.
+
+  $c +cc -cc prcpal Arg .cc invc $.
+
+  $( Syntax for the function ` prcpal ` . $)
+  cprcpal $a class prcpal $.
+
+  $( Define the function ` prcpal ` .  (Contributed by BJ, 22-Jun-2019.) $)
+  df-bj-prcpal $a |- prcpal = ( x e. RR |-> ( ( x mod ( 2 x. _pi ) ) -
+                 if ( ( x mod ( 2 x. _pi ) ) <_ _pi , 0 , ( 2 x. _pi ) ) ) ) $.
```

Comment:
With this definition, you will probably have to handle many cases. Maybe it would be more straightforward if you define it as ` Im(log(exp(iθ))) ` (where `θ` is your `x`). That way, you could benefit of all the properties and theorems already available for ` Im(log x) `.